src/resource/solidDataset.test.ts

```diff
@@ -3534,103 +3534,205 @@ describe("getWellKnownSolid", () => {
+    expect(mockedFetcher.fetch.mock.calls[0][0]).toBe(
+      serverUrl.concat(wellKnownSolid)
+    );
+    // Checks the resource's location header otherwise
+    expect(mockedFetcher.fetch.mock.calls[1][0]).toBe(resourceUrl);
+    // The advertised podIdentifier (as storage) is used to determine well known Solid
+    expect(mockedFetcher.fetch.mock.calls[2][0]).toBe(
+      podUrl.concat(wellKnownSolid)
     );
   });
 
-  it("assumes the discovery resource is at the server's origin if the resource is not found", async () => {
+  it("uses the given fetcher for pod root well known solid if provided", async () => {
+    // Root cannot be fetched
+    mockedFetcher.fetch.mockResolvedValueOnce(
+      new Response(undefined, { status: 404 })
+    );
     const mockFetch = jest.fn(window.fetch);
-    setMockResourceResponseOnFetch(mockFetch).mockResolvedValueOnce(
-      new Response("", {
-        status: 404,
-        statusText: "Not found",
+    // Resource advertises Pod root
+    mockFetch.mockResolvedValueOnce(
+      mockResponse(undefined, {
+        url: resourceUrl,
+        headers: {
+          "Content-Type": "text/turtle",
+          link: `<${podUrl}>; rel="http://www.w3.org/ns/pim/space#storage"`,
+        },
       })
     );
     setMockWellKnownSolidResponseOnFetch(mockFetch);
 
-    await getWellKnownSolid("https://some.pod/resource/x/y/z", {
-      fetch: mockFetch,
-    });
+    await getWellKnownSolid(resourceUrl, { fetch: mockFetch });
 
-    expect(mockFetch.mock.calls[0][0]).toBe("https://some.pod/resource/x/y/z");
-    expect(mockFetch.mock.calls[2][0]).toBe(
-      "https://some.pod/.well-known/solid"
+    // Fails at pod root (unauthenticated)
+    expect(mockedFetcher.fetch.mock.calls[0][0]).toBe(
+      serverUrl.concat(wellKnownSolid)
     );
+    // Checks the pod root (authenticated/with the provided fetcher)
+    expect(mockFetch.mock.calls[0][0]).toBe(resourceUrl);
+    // Retrieve pod root well known solid
+    expect(mockFetch.mock.calls[1][0]).toBe(podUrl.concat(".well-known/solid"));
   });
 
-  it("forms the .well-known/solid endpoint from the root resource", async () => {
-    const mockFetch = jest.fn(window.fetch);
-    setMockResourceResponseOnFetch(mockFetch);
-    setMockWellKnownSolidResponseOnFetch(mockFetch);
+  it("appends a / to the Pod root if missing before appending .well-known/solid", async () => {
+    // Root cannot be fetched
+    mockedFetcher.fetch.mockResolvedValueOnce(
+      new Response(undefined, { status: 404 })
+    );
+    // Resource advertises Pod root
+    mockedFetcher.fetch.mockResolvedValueOnce(
+      mockResponse(undefined, {
+        url: resourceUrl,
+        headers: {
+          "Content-Type": "text/turtle",
+          link: `</username>; rel="http://www.w3.org/ns/pim/space#storage"`,
+        },
+      })
+    );
+    // Fetches Pod root well known
+    mockedFetcher.fetch.mockResolvedValueOnce(
+      new Response(
```

Comment:
Could we store these responses in variables ahead of time? That or in a fixtures directory?

Reply:
I would rather be able to read the tests easily ^^,